scripts/test/SANS/command_interface/batch_csv_file_parser_test.py

```diff
@@ -54,12 +54,16 @@ def test_that_raises_when_sample_scatter_is_missing(self):
         self.assertRaises(RuntimeError, parser.parse_batch_file)
         BatchCsvParserTest._remove_csv(batch_file_path)
 
-    def test_that_raises_when_output_is_missing(self):
+    def test_that_does_not_raise_when_output_is_missing(self):
         content = "# MANTID_BATCH_FILE add more text here\n" \
                    "sample_sans,test,output_as,\n"
         batch_file_path = BatchCsvParserTest._save_to_csv(content)
         parser = BatchCsvParser(batch_file_path)
-        self.assertRaises(RuntimeError, parser.parse_batch_file)
+        try:
+            parser.parse_batch_file()
+        except RuntimeError as e:
+            self.assertTrue(False, "Did not expect a RuntimeError to be raised when parsing batch file without an "
```

Comment:
This could use [`self.fail()`](https://docs.python.org/3.5/library/unittest.html#unittest.TestCase.fail).

Reply:
In fact the `try` and assertions is redundant here. If an exception is raised the test will fail regardless.

Reply:
I think using self.fail and having a clearer failure message would be a better way to handle this test. This will allow us to distinguish between a failure due to a RuntimeError which has a change of being re-introduced through developer error and other errors.